### PR TITLE
Crc/custom component multi field

### PR DESCRIPTION
**Pull request by VilliGunn**
# \<Description\>

https://app.asana.com/0/1199733748699731/1199914212397854

- Added an option to have multiple named fields in one custom component instead of array of inputs.
- All form errors are now accessible to the component rather than just one error per comonent.

## What

- Added `childInputIds` to the `buildCustomField` function
- In the `extractAnswersToSubmitFromScreen` we now check if the `screen.type` is custom and if the `childInputIds` is available. If so then we use the array instead of the `screen.id`

## Why

To have better control of the answers in custom components with multiple field inputs.

## Screenshots / Gifs

Screenshot 2021-02-23 at 14 00 25

## Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] Formatting passes locally with my changes
- [x] I have rebased against main before asking for a review

### CODE REVIEW

File: libs/application/templates/children-residence-change/src/fields/ContactInfo/index.tsx [hunk 1]
```diff
@@ -30,22 +35,24 @@ const ContactInfo = ({ error, application, field }: FieldBaseProps) => {
       </Box>
       <Box marginTop={5}>
         <Controller
-          name="email"
-          defaultValue={getValue('contactInfo')?.[0]}
+          name={emailId}
+          defaultValue={getValue(phoneNumberId)}
```
Comment (Rikkiabb) [inline libs/application/templates/children-residence-change/src/fields/ContactInfo/index.tsx:39]: Should this be `emailId` ?

File: libs/application/templates/children-residence-change/src/fields/ContactInfo/index.tsx [hunk 2]
```diff
@@ -30,22 +35,24 @@ const ContactInfo = ({ error, application, field }: FieldBaseProps) => {
...
+                ref={register}
+                id={emailId}
+                name={emailId}
                 backgroundColor="blue"
-                type="text"
+                type="email"
                 label={formatMessage(otherParent.inputs.emailLabel)}
                 value={value}
-                hasError={!!error}
+                hasError={!!emailError}
```
Comment (Rikkiabb) [inline libs/application/templates/children-residence-change/src/fields/ContactInfo/index.tsx:50]: Do we have to use not not here? Wondering if we know the type `undefined | string` and can then check if it is undefined or do we ever get an empty string?

File: libs/application/templates/children-residence-change/src/fields/ContactInfo/index.tsx [hunk 3]
```diff
@@ -6,15 +6,20 @@ import { Controller, useFormContext } from 'react-hook-form'
 import { otherParent } from '../../lib/messages'
 import { extractParentFromApplication } from '../../lib/utils'
 
-const ContactInfo = ({ error, application, field }: FieldBaseProps) => {
-  const { id } = field
+const emailId = 'email'
+const phoneNumberId = 'phoneNumber'
+
+export const contactInfoIds = [emailId, phoneNumberId]
```
Comment (Rikkiabb) [inline libs/application/templates/children-residence-change/src/fields/ContactInfo/index.tsx:12]: Like the approach that the component "owns" the id's

**Review by Rikkiabb — COMMENTED** [review]
Small questions from me :) Nice stuff 👏

File: libs/application/templates/children-residence-change/src/fields/ContactInfo/index.tsx (hunk 1 above)
Comment (VilliGunn) [reply libs/application/templates/children-residence-change/src/fields/ContactInfo/index.tsx:39]: I though about it but decided to go with the same approach as all the wrapper controller comopnents - same name for the controller and the input fields:
https://github.com/island-is/island.is/blob/main/libs/shared/form-fields/src/lib/InputController/InputController.tsx#L143

File: libs/application/templates/children-residence-change/src/fields/ContactInfo/index.tsx (hunk 2 above)
Comment (VilliGunn) [reply libs/application/templates/children-residence-change/src/fields/ContactInfo/index.tsx:50]: No we shouldn't get an empty string here. Always just undefined or some error value

**Review by Rikkiabb — APPROVED** [review]
Nice stuff 👏

**Review by jeremybarbet — APPROVED** [review]
LGTM

**Pull request merged**